src/Build/Instance/TaskRegistry.cs

```diff
@@ -1425,6 +1431,10 @@ private bool GetTaskFactory(TargetLoggingContext targetLoggingContext, ElementLo
                                     return false;
                                 }
                             }
+                            catch(InvalidProjectFileException)
+                            {
+                                throw;
```

Comment:
Okay I've removed the unique error code and thrown an `InvalidOperationException` so the overall error contains the message

```
nulltasktype.proj(10,5): error MSB4175: The task factory "Microsoft.Build.UnitTests.BackEnd.TaskRegistry_Tests+NullTaskTypeTaskFactory" could not be loaded from the assembly "D:\msbuild\bin\Debug\AnyCPU\Windows_NT\Output\Microsoft.Build.Engine.UnitTests.dll".  The task factory must return a value for the "TaskType" property.
```